src/app/drive/services/folder.service.ts

```diff
@@ -349,39 +356,62 @@ export async function downloadFolderAsZip({
     }
   }, maxConcurrency);
 
+  const addUniqueItem = (item) => {
+    const isItemAlreadyAdded = totalItems.some((i) => i.id === item.id);
+    if (!isItemAlreadyAdded) {
+      totalItems.push(item);
+    }
+  };
+
   const downloadFolder = async (folderToDownload: FolderRef) => {
     const files = await addAllFilesToZip(
       folderToDownload.name,
       async (file) => {
-        updateNumItems();
-        const lruFilesCacheManager = await LRUFilesCacheManager.getInstance();
-        const cachedFile = await lruFilesCacheManager.get(file.uuid?.toString());
-        const isCachedFileOlder = checkIfCachedSourceIsOlder({ cachedFile, file });
-
-        if (cachedFile?.source && !isCachedFileOlder) {
-          return cachedFile.source.stream();
+        addUniqueItem(file);
+        // TODO: QA REMOVE
+        if (file.name.includes('internxt_test_file') || file.plainName?.includes('internxt_test_file')) {
+          failedItems.push(file as DriveFileData);
+          return;
+        }
+        try {
+          updateNumItems();
+          const lruFilesCacheManager = await LRUFilesCacheManager.getInstance();
+          const cachedFile = await lruFilesCacheManager.get(file.uuid?.toString());
+          const isCachedFileOlder = checkIfCachedSourceIsOlder({ cachedFile, file });
+
+          if (cachedFile?.source && !isCachedFileOlder) {
+            return cachedFile.source.stream();
+          }
+
+          const downloadedFileStream = await downloadFile({
+            bucketId: file.bucket,
+            fileId: file.fileId,
+            creds: options.credentials,
+            mnemonic: options.mnemonic,
+            options: {
+              notifyProgress: () => {},
+              abortController,
+            },
+          });
+
+          const sourceBlob = await binaryStreamToBlob(downloadedFileStream, file.type || '');
+          await updateDatabaseFileSourceData({
+            folderId: file.folderId,
+            sourceBlob,
+            fileId: file.id,
+            updatedAt: file.updatedAt,
+          });
+
+          return sourceBlob.stream();
+        } catch (error: any) {
+          const serverUnavailableError = error.message === 'Server unavailable';
```

Comment:
by convention, the boolean variables is starts with prefixes as  `is`, `has`, `can`, `should`, etc.
Comparing the message in this way is very susceptible to not being correct with respect to changes, if you want to check the message I think it would be better to use an includes, as it provides a bit more flexibility, although this kind of comparison is not the best. If possible it is better to use the error codes.